src/cli/qcmp.rs

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+use std::net::SocketAddr;
+
+use crate::protocol::Protocol;
+
+#[derive(Clone, Debug, clap::Subcommand)]
+pub enum Qcmp {
+    Ping(Ping),
+}
+
+/// Pings a endpoint for a `amount` of attempts, printing the resulting
+/// median, average, and number of successful attempts.
+#[derive(clap::Args, Clone, Debug)]
+pub struct Ping {
+    /// The quilkin endpoint to ping
+    pub endpoint: SocketAddr,
+    /// The number of pings to send to the endpoint (default: 5).
+    #[clap(short, long, default_value_t = 5)]
+    pub amount: usize,
+}
+
+impl Ping {
+    pub async fn run(&self) -> crate::Result<()> {
+        tracing::info!("starting ping task");
+        let socket = tokio::net::UdpSocket::bind((std::net::Ipv4Addr::UNSPECIFIED, 0)).await?;
```

Comment:
Since you know the `SocketAddr` should check if it's ipv4 or ipv6 and use the appropriate `UNSPECIFIED`.

(not sure if you can go from v6 ->, v6?)

Could possibly also wait for #666, which I'm working on right now, but since it's not listening for a long time, it's probably overkill.